datreant/backends/statefiles.py

```diff
@@ -0,0 +1,448 @@
+"""
+Interface classes for state files.
+
+"""
+
+import os
+import warnings
+import json
+
+from six import string_types
+
+import datreant
+from .core import FileSerial
+
+
+class MixinJSON(object):
+    def _deserialize(self, handle):
+        return json.load(handle)
+
+    def _serialize(self, record, handle):
+        json.dump(record, handle)
+
+
+class TreantFile(MixinJSON, FileSerial):
+    def __init__(self, filename, logger=None, **kwargs):
+        """Initialize Treant state file.
+
+        This is the base class for all Treant state files. It generates data
+        structure elements common to all Treants. It also implements
+        low-level I/O functionality.
+
+        :Arguments:
+            *filename*
+                path to file
+            *logger*
+                Treant's logger instance
+
+        :Keywords:
+            *treanttype*
+                Treant type
+            *name*
+                user-given name of Treant object
+            *categories*
+                user-given dictionary with custom keys and values; used to
+                give distinguishing characteristics to object for search
+            *tags*
+                user-given list with custom elements; used to give
+                distinguishing characteristics to object for search
+            *version*
+                version of datreant file was generated with
+
+        .. Note:: kwargs passed to :meth:`create`
+
+        """
+        super(FileSerial, self).__init__(filename, logger=logger)
+
+        # if file does not exist, it is created; if it does exist, it is
+        # updated
+        try:
+            self.create(**kwargs)
+        except OSError:
+            # in case the file exists but is read-only; we can't update but may
+            # still want to use it
+            if os.path.exists(self.filename):
+                pass
+            # if the file doesn't exist, we still want an exception
+            else:
+                raise
+
+    def _init_record(self):
```

Comment:
if not then it might be better to move this code to the `__init__` function as well.